packages/block-editor/src/components/inserter/menu.native.js

```diff
@@ -201,6 +203,7 @@ function InserterMenu( {
 			hasNavigation
 			setMinHeightToMaxHeight={ showSearchForm }
 			contentStyle={ styles.list }
+			isFullScreen={ ! isIOS && showSearchForm }
```

Comment:
There are cases ( Buttons block for instance) where the inserter menu only has a few blocks. In these cases we don't show the search input and thus do not need to go full screen on the menu